### PR TITLE
docs: DOC-205: Document changes to ML backend

### PR DESCRIPTION
**Pull request by caitlinwheeless**
Updates to ML backend 

This affects:

- [X] Community docs
- [X] Enterprise docs

### CODE REVIEW

File: docs/source/guide/active_learning.md [hunk 1]
```diff
@@ -13,25 +13,22 @@ section: "Machine learning"
...
+Creating annotated training data for supervised machine learning models can be expensive and time-consuming. Active Learning is a branch of machine learning that seeks to **minimize the total amount of data required for labeling by strategically sampling observations** that provide new insight into the problem. 
+
+In particular, Active Learning algorithms aim to select diverse and informative data for annotation, rather than random observations, from a pool of unlabeled data using **prediction scores**. For more about the practice of active learning, read [this article written by our HumanSignal CTO on Towards Data Science](https://towardsdatascience.com/learn-faster-with-smarter-data-labeling-15d0272614c4).
```
Comment (niklub) [inline docs/source/guide/active_learning.md:23]: always forgot to fix an important typo mentioned in the comments to this article, namely, replacing `max` -> `min` in this formula:
image
can we do that at some point?

File: docs/source/guide/active_learning.md [hunk 2]
```diff
@@ -13,25 +13,22 @@ section: "Machine learning"
...
 
 ## Set up an automated active learning loop
 
 Continuously train and review predictions from a connected machine learning model using Label Studio. 
 
 <br/><img src="/images/LS-active-learning.jpg" alt="Diagram of the active learning workflow described in surrounding text" class="gif-border" width="800px" height="472px" />
 
-After a user creates an annotation in Label Studio, the configured webhook sends a message to the machine learning backend with the information about the created annotation. The fit() method of the ML backend runs to train the model. When the user moves on to the next labeling task, Label Studio retrieves the latest prediction for the task from the ML backend, which runs the predict() method on the task.
+After a user creates an annotation in Label Studio, the configured webhook sends a message to the machine learning backend with the information about the created annotation. The `fit()` method of the ML backend runs to train the model. When the user moves on to the next labeling task, Label Studio retrieves the latest prediction for the task from the ML backend, which runs the predict() method on the task.
```
Comment (niklub) [inline docs/source/guide/active_learning.md]:
```suggestion
After a user creates an annotation in Label Studio, the configured webhook sends a message to the machine learning backend with the information about the created annotation. The `fit()` method of the ML backend runs to train the model. When the user moves on to the next labeling task, Label Studio retrieves the latest prediction for the task from the ML backend, which runs the `predict()` method on the task.
```

File: docs/source/guide/active_learning.md [hunk 3]
```diff
@@ -49,7 +46,7 @@ As you label tasks, Label Studio sends webhook events to your machine learning b
 ## Connect the ML backend to Label Studio for active learning
 
 1. Follow the steps to [Add an ML backend to Label Studio](ml.html#Add-an-ML-backend-to-Label-Studio).
-2. Under **ML-Assisted Labeling**, enable the setting to **Show predictions to annotators in the Label Stream and Quick View**. 
+2. Under **Model**, enable the setting to **Start model training on annotation submission**.
```
Comment (niklub) [inline docs/source/guide/active_learning.md]: let's remove this line - it is already the default behavior. Worth mentioning though, that it sends "training" request to ML backend after each annotation submission or update

File: docs/source/guide/ml.md [hunk 4]
```diff
@@ -10,97 +10,148 @@ meta_description: Machine learning frameworks for integrating your model develop
...
 ---
 
-Integrate your model development pipeline with your data labeling workflow by adding a machine learning (ML) backend to Label Studio. You can set up your favorite machine learning frameworks to do the following:
-- **Pre-labeling** by letting models predict labels and then have annotators perform further manual refinements. 
-- **Auto-labeling** by letting models create automatic annotations. 
-- **Online Learning** by simultaneously updating your model while new annotations are created, letting you retrain your model on-the-fly. 
-- **Active Learning** by selecting example tasks that the model is uncertain how to label for your annotators to label manually.
+You can use an ML backend to integrate your model development pipeline with your data labeling workflow. There are several use cases, including: 
+
+- Pre-annotate data with a model
```
Comment (niklub) [inline docs/source/guide/ml.md]: Pre-annotate, or Autolabel data with model

File: docs/source/guide/ml.md [hunk 5]
```diff
@@ -10,97 +10,148 @@ meta_description: Machine learning frameworks for integrating your model develop
...
+- Model fine-tuning based on recently annotated data
 
 For example, for an image classification task, the model pre-selects an image class for data annotators to verify. For audio transcriptions, the model displays a transcription that data annotators can modify. 
 
-!!! note
-    Use [Label Studio Enterprise to build an automated active learning loop](https://docs.humansignal.com/guide/active_learning.html) with a machine learning model backend. If you use the open source Community Edition of Label Studio, you can manually sort tasks and retrieve predictions to mimic an active learning process.
+!!! info Tip
+    You can use [Label Studio Enterprise to build an automated active learning loop](https://docs.humansignal.com/guide/active_learning.html) with a machine learning model backend. If you use the open source Community Edition of Label Studio, you can manually sort tasks and retrieve predictions to mimic an active learning process.
 
+If you just need to load static pre-annotated data into Label Studio, running an ML backend might be overkill for you. Instead, you can [import preannotated data](predictions).
```
Comment (niklub) [inline docs/source/guide/ml.md:28]: I feel before this line, we have to explain how it works:
1. User opens the task
2. LS sends the request to ML backend
3. ML backend return the online prediction response
4. LS shows prelabeling on the screen

File: docs/source/guide/ml.md [hunk 6]
```diff
@@ -10,97 +10,148 @@ meta_description: Machine learning frameworks for integrating your model develop
...
+
+!!! info Tip
     You can also [add an ML backend using the API](/api/#operation/api_ml_create). You will need the project ID and the machine learning backend URL. 
-   
-## Train a model
 
-After you connect a model to Label Studio as a machine learning backend and annotate at least one task, you can start training the model. 
+## Example models
+
+For a list of the available example models, see the [Models table in GitHub](https://github.com/HumanSignal/label-studio-ml-backend?tab=readme-ov-file#models).
```
Comment (niklub) [inline docs/source/guide/ml.md]: Why don't we want to move the full table here for better visibility? or is it planned for the future?

File: docs/source/guide/ml.md [hunk 7]
```diff
@@ -10,97 +10,148 @@ meta_description: Machine learning frameworks for integrating your model develop
...
+* [Label Studio](install_enterprise)
+* [Docker Compose](https://docs.docker.com/compose/install/)
+
+</div>
+
+### Start the model
+
+1. First, decide which [model](#Example-models) you want to use and check for required parameters (click the link for each model to see a full parameter list). 
+
+    Use [`label_studio_ml/default_configs/docker-compose.yml`](https://github.com/HumanSignal/label-studio-ml-backend/blob/master/label_studio_ml/default_configs/docker-compose.yml) for general configuration, and `docker-compose.yml` within the model directory for model-specific parameters.
```
Comment (niklub) [inline docs/source/guide/ml.md]: we don't need to refer general configuration here - always use `docker-compose.yml` from the model directory, it contains all configuration options.
You only have to know about `default_configs` when you start creating your own ML backend

File: docs/source/guide/ml.md [hunk 8]
```diff
@@ -10,97 +10,148 @@ meta_description: Machine learning frameworks for integrating your model develop
...
-   ```
-1. Change to the directory with the Docker Compose configuration file:
-   ```bash
-   cd label-studio-ml-backend/label_studio_ml/examples/the_simplest_ml_backend
-   ```
-2. Start Docker Compose:
     ```bash
-    docker-compose up
+git clone https://github.com/HumanSignal/label-studio-ml-backend.git
+cd label-studio-ml-backend/label_studio_ml/examples/{MODEL_NAME}
```
Comment (niklub) [inline docs/source/guide/ml.md]:
```suggestion
cd label-studio-ml-backend/label_studio_ml/examples/segment_anything_model
```

File: docs/source/guide/ml.md [hunk 9]
```diff
@@ -10,97 +10,148 @@ meta_description: Machine learning frameworks for integrating your model develop
...
+
+### Start the model
+
+1. First, decide which [model](#Example-models) you want to use and check for required parameters (click the link for each model to see a full parameter list). 
+
+    Use [`label_studio_ml/default_configs/docker-compose.yml`](https://github.com/HumanSignal/label-studio-ml-backend/blob/master/label_studio_ml/default_configs/docker-compose.yml) for general configuration, and `docker-compose.yml` within the model directory for model-specific parameters. 
+
+2. Then replace `{MODEL_NAME}` in the below command with the appropriate directory. 
+
+    For example, if you are using the SAM backend, the model name would be `segment_anything_model`, which matches the [directory name in the repo](https://github.com/HumanSignal/label-studio-ml-backend/tree/master/label_studio_ml/examples/segment_anything_model):
```
Comment (niklub) [inline docs/source/guide/ml.md]:
```suggestion
    For example, if you are using the SAM backend with [SegmentAnything model](https://segment-anything.com/), the model name would be `segment_anything_model`, which matches the [directory name in the repo](https://github.com/HumanSignal/label-studio-ml-backend/tree/master/label_studio_ml/examples/segment_anything_model):
```

File: docs/source/guide/ml.md [hunk 10]
```diff
@@ -10,97 +10,148 @@ meta_description: Machine learning frameworks for integrating your model develop
...
-    docker-compose up
+git clone https://github.com/HumanSignal/label-studio-ml-backend.git
+cd label-studio-ml-backend/label_studio_ml/examples/{MODEL_NAME}
+docker-compose up
     ```
 
-You can access the machine learning backend server at `http://localhost:9090`. You can also use this URL to add the machine learning backend to Label Studio (see below).
+The model should begin running at `http://localhost:9090`. You can verify this by clicking **Send Test Request** from the overflow menu next to the model or by using the following command: 
+
+`curl http://localhost:9090/health`
```
Comment (niklub) [inline docs/source/guide/ml.md]:
```suggestion
```bash
> curl http://localhost:9090
{"model_class":"SamMLBackend","status":"UP"}
```
The healthy output contains `"status"` field and the running model name
```

File: docs/source/guide/ml.md [hunk 11]
```diff
@@ -10,97 +10,148 @@ meta_description: Machine learning frameworks for integrating your model develop
...
 
-!!! note
+</div>
+
+Click **Connect Model** and complete the following fields:
+
+| Field | Description                                                                            |
+| -------- | -------------------------------------------------------------------------------------- |
+| **Name**   | Enter a name for the model.                        |
+| **Backend URL**  | Enter a URL for the model. <br /><br />If you are following the steps above, this would be `http://localhost:9090`.  |
```
Comment (niklub) [inline docs/source/guide/ml.md]: We should add the note about `localhost` can be inaccessible if LS is running from docker (as it has been mentioned above)

File: docs/source/guide/ml.md [hunk 12]
```diff
@@ -10,97 +10,148 @@ meta_description: Machine learning frameworks for integrating your model develop
...
+| **Select authentication method**   | If a username and password are required to access the model, you can select **Basic Authentication** and enter them here.                                     |
+| **Extra params**  | Enter any additional parameters you want to pass to the model.                                      |
+| **Interactive preannotations**  | Enable this option to allow the model to assist with the labeling process by providing real-time predictions or suggestions as annotators work on tasks.  <br /><br />In other words, as you interact with data (for example, by drawing a region on an image, highlighting text, or asking an LLM a question), the ML backend receives this input and returns predictions based on it.  |
```
Comment (niklub) [inline docs/source/guide/ml.md]: should be the link to interactive labeling guide here

File: docs/source/guide/ml.md [hunk 13]
```diff
@@ -10,97 +10,148 @@ meta_description: Machine learning frameworks for integrating your model develop
...
+
+</div>
+
+<div class="opensource-only">
+
+From the [**Model** page](project_settings#Model) under project settings, select one of the following: 
+
+</div>
+
+* **Start model training on annotation submission**--Enable this option for automated training. When enabled, training is automatically initiated every time an annotation is submitted or updated.
```
Comment (niklub) [inline docs/source/guide/ml.md]: Please remove this line

File: docs/source/guide/ml.md [hunk 14]
```diff
@@ -10,97 +10,148 @@ meta_description: Machine learning frameworks for integrating your model develop
...
+</div>
+
+<div class="opensource-only">
+
+From the [**Model** page](project_settings#Model) under project settings, select one of the following: 
+
+</div>
+
+* **Start model training on annotation submission**--Enable this option for automated training. When enabled, training is automatically initiated every time an annotation is submitted or updated. 
+* **Start Training** (Available from the overflow menu next to the connected model)--Manually initiate training. Use this action if you want to control when the model training occurs, such as after a specific number of annotations have been collected or at certain intervals.
```
Comment (niklub) [inline docs/source/guide/ml.md]:
```suggestion
* **Start Training** (Available from the overflow menu next to the connected model) -- Manually initiate training. Use this action if you want to control when the model training occurs, such as after a specific number of annotations have been collected or at certain intervals.
```

File: docs/source/guide/ml.md [hunk 15]
```diff
@@ -10,97 +10,148 @@ meta_description: Machine learning frameworks for integrating your model develop
...
+* From the API, specify the ID of the machine learning backend and run the following command: 
    ```
    curl -X POST http://localhost:8080/api/ml/{id}/train
    ```
   See [the Train API documentation](/api/#operation/api_ml_train_create) for more.
-- [Trigger training with webhooks](ml_create.html#Trigger-training-with-webhooks).
+- [Trigger training with webhooks](ml_create#Trigger-training-with-webhooks).
 
-In development mode, training logs appear in the web browser console. In production mode, you can find runtime logs in `my_backend/logs/uwsgi.log` and RQ training logs in `my_backend/logs/rq.log` on the server running the ML backend, which might be different from the Label Studio server. To see more detailed logs, start the ML backend server with the `--debug` option. 
+In development mode, training logs appear in the web browser console. In production mode, you can find runtime logs in `my_backend/logs/uwsgi.log` and RQ training logs in `my_backend/logs/rq.log` on the server running the ML backend, which might be different from the Label Studio server.
```
Comment (niklub) [inline docs/source/guide/ml.md]: it is obsolete - no wsgi/rq processes exist. All training logs stream in stdout / console

File: docs/source/guide/ml.md [hunk 16]
```diff
@@ -10,97 +10,148 @@ meta_description: Machine learning frameworks for integrating your model develop
...
+!!! note
+    The terms "predictions" and pre-annotations" are used interchangeably. 
+
+
+### Get predictions from a model
+
+After you connect a model to Label Studio, you can see model predictions in the labeling interface if the model is pre-trained, or right after it finishes [training](#Model-training). 
+
+* To manually add predictions, go to the Data Manager, select the tasks you want to get predictions for, and then select **Actions > Retrieve predictions**.
+* To automatically pre-label data with predictions, go to the project settings and enable **Annotation > Use predictions to prelabel data**.
```
Comment (niklub) [inline docs/source/guide/ml.md]: we should add the note:
"ensure the connected model is selected in dropdown `Select which predictions or which model you want to use:`"

File: docs/source/guide/ml.md [hunk 17]
```diff
@@ -114,29 +165,16 @@ If you want to retrieve predictions manually for a list of tasks **using only an
...
 
-1. In the Label Studio UI, open the project that you want to use with your ML backend.
-2. Select **Settings > Machine Learning**.
-3. For the connected model you wish to enable, click **Edit** in the overflow menu.
-4. Enable **Use for interactive preannotation**.
-5. Click **Validate and Save**.
+#### Smart tools
 
-For image labeling, you can update your labeling configuration to include `smart="true"` option for the type of labeling you're performing. Smart tools appear by default if auto-annotation is enabled. If you only want the smart option to appear and don't want to perform manual labeling at all, use `smartOnly="true"`. 
+For image labeling, you can update your labeling configuration to include the `smart="true"` option for the type of labeling you're performing. Smart tools appear by default if auto-annotation is enabled. If you only want the smart option to appear and don't want to perform manual labeling at all, use `smartOnly="true"`.
```
Comment (niklub) [inline docs/source/guide/ml.md]: It seems we need to elaborate on what smart tools are:
- tools that duplicate controls to allow interaction with connected ML backend (currently supported for <RectangleLabels>, <Keypoints>, <BrushLabels>, <Ellipse>, <POlygon>)
- give the data in `context` variable in ML backend /predict request + link to explanation what the context variable is and how to use it

File: docs/source/guide/ml.md [hunk 18]
```diff
@@ -165,13 +203,19 @@ curl -H 'Authorization: Token <user-token-from-account-page>' -X POST \
 
 ### Choose which predictions to display to annotators
 
-You can choose which model predictions to display to annotators by default. 
+<div class="opensource-only">
+
+You can choose which model or prediction set to display to annotators by default. This is available under the **Annotation > Pre-labeing** in the project settings.
```
Comment (niklub) [inline docs/source/guide/ml.md]:
```suggestion
You can choose which model or prediction set to display to annotators by default. This is available under the **Annotation > Live Predictions** in the project settings. 
```

File: docs/source/guide/ml.md [hunk 19]
```diff
@@ -165,13 +203,19 @@ curl -H 'Authorization: Token <user-token-from-account-page>' -X POST \
...
-You can choose which model predictions to display to annotators by default. 
+<div class="opensource-only">
+
+You can choose which model or prediction set to display to annotators by default. This is available under the **Annotation > Pre-labeing** in the project settings. 
+
+</div>
+
+<div class="enterprise-only">
+
+You can choose which model or prediction set to display to annotators by default. This is available under the **Annotation > Predictions** in the project settings.
```
Comment (niklub) [inline docs/source/guide/ml.md]:
```suggestion
You can choose which model or prediction set to display to annotators by default. This is available under the **Annotation > Live Predictions** in the project settings. 
```

File: docs/source/guide/ml_create.md [hunk 20]
```diff
@@ -5,120 +5,57 @@ type: guide
...
+meta_title: Write your own ML backend
 meta_description: Set up your machine learning model to output and consume predictions in your data science and data labeling projects. 
 section: "Machine learning"
 
 ---
 
 Use the Label Studio ML backend to integrate Label Studio with machine learning models. The Label Studio ML backend is an SDK that you can use to wrap your machine learning model code and turn it into a web server. The machine learning server uses [uWSGI](https://uwsgi-docs.readthedocs.io/en/latest/) and [supervisord](http://supervisord.org/), and handles background training jobs with [RQ](https://python-rq.org/).
 
-There are several use cases for the ML backend:
+Follow the steps below to wrap custom machine learning model code with the Label Studio ML SDK, or see [our library of example ML backends](ml_tutorials.html) to integrate with popular machine learning frameworks such as PyTorch, GPT2, and others.
```
Comment (niklub) [inline docs/source/guide/ml_create.md]:
```suggestion
Follow the steps below to wrap custom machine learning model code with the Label Studio ML SDK, or see [our library of example ML backends](ml_tutorials.html) to integrate with popular machine learning frameworks and tools such as [Huggingface's Transformers](https://huggingface.co/docs/transformers/index), [OpenAI](https://openai.com/), [Langchain](https://www.langchain.com/) and others. 
```

File: docs/source/guide/ml_create.md [hunk 21]
```diff
@@ -5,120 +5,57 @@ type: guide
...
-```bash
-label-studio-ml start my_ml_backend -p 9091
 ```
+my_ml_backend/
+├── Dockerfile
+├── docker-compose.yml
+├── model.py
+├── _wsgi.py
+├── README.md
+└── requirements.txt
```
Comment (niklub) [inline docs/source/guide/ml_create.md]:
```suggestion
my_ml_backend/
├── Dockerfile
├── .dockerignore
├── docker-compose.yml
├── model.py
├── _wsgi.py
├── README.md
├── requirements-base.txt
├── requirements-test.txt
├── requirements.txt
└── test_api.py
```

File: docs/source/guide/ml_create.md [hunk 22]
```diff
@@ -5,120 +5,57 @@ type: guide
...
+```
+
+Where:
+
+* `Dockerfile` and `docker-compose.yml` are used to run the ML backend with Docker.
```
Comment (niklub) [inline docs/source/guide/ml_create.md]:
```suggestion
* `Dockerfile`, `docker-compose.yml` and `.dockerignore` are used to run the ML backend with Docker. 
```

File: docs/source/guide/ml_create.md [hunk 23]
```diff
@@ -5,120 +5,57 @@ type: guide
...
+├── README.md
+└── requirements.txt
+```
+
+Where:
+
+* `Dockerfile` and `docker-compose.yml` are used to run the ML backend with Docker. 
+* `model.py` is the main file where you can implement your own training and inference logic. 
+* `_wsgi.py` is a helper file that is used to run the ML backend with Docker (you don't need to modify this). 
+* `README.md` has instructions on how to run the ML backend.
```
Comment (niklub) [inline docs/source/guide/ml_create.md]:
```suggestion
* `README.md` must contain instructions on how to run the ML backend. 
```

File: docs/source/guide/ml_create.md [hunk 24]
```diff
@@ -5,120 +5,57 @@ type: guide
...
+└── requirements.txt
+```
+
+Where:
+
+* `Dockerfile` and `docker-compose.yml` are used to run the ML backend with Docker. 
+* `model.py` is the main file where you can implement your own training and inference logic. 
+* `_wsgi.py` is a helper file that is used to run the ML backend with Docker (you don't need to modify this). 
+* `README.md` has instructions on how to run the ML backend. 
+* `requirements.txt` is a file with Python dependencies.
```
Comment (niklub) [inline docs/source/guide/ml_create.md]:
```suggestion
* `requirements.txt` is where you put your Python dependencies.
* `requirements_base.txt` and `requirements_test.txt` are basic dependencies (you don't need to modify this)
* `test_api.py` is where you put your model tests
```

File: docs/source/guide/ml_create.md [hunk 25]
```diff
@@ -130,14 +67,13 @@ def predict(self, tasks, context, **kwargs):
...
 
-Once you implement the `predict()` method, you can see predictions from the connected ML backend in Label Studio.
+- `tasks`: [Label Studio tasks in JSON format](task_format)
+- `context`: [Label Studio context in JSON format](#Support-interactive-pre-annotations-in-your-ML-backend) - for an interactive labeling scenario
+- `predictions`: [Predictions array in JSON format](export#Raw-JSON-format-of-completed-tasks)
 
-For another example of the `predict()` method, see [model.py](https://github.com/HumanSignal/label-studio-ml-backend/blob/master/label_studio_ml/examples/the_simplest_backend/model.py#L22).
+Once you implement the `predict` method, you can see predictions from the connected ML backend in Label Studio.
 
 ### Support interactive pre-annotations in your ML backend
```
Comment (niklub) [inline docs/source/guide/ml_create.md:84]: [the url below](https://github.com/HumanSignal/label-studio/pull/5642/files#diff-9e871e6b36048f0dcb8692fcb33c3fc4ca77e2b26e7cc36a24dca5c8aca6eefaL144) has to be changed to 

https://github.com/heartexlabs/label-studio-ml-backend/tree/master/label_studio_ml/examples/interactive_substring_matching

File: docs/source/guide/ml_create.md [hunk 26]
```diff
@@ -177,30 +114,72 @@ def fit(self, event, data, **kwargs):
...
 
 ### Other methods and parameters
-Other methods and parameters available within the `LabelStudioMLBase` class include:
 
-- `self.label_config` - returns the [Label Studio labeling config](https://labelstud.io/guide/setup.html) as XML string.
-- `self.parsed_label_config` - returns the [Label Studio labeling config](https://labelstud.io/guide/setup.html) as JSON.
-- `self.model_version` - returns the current model version.
+Other methods and parameters are available within the `LabelStudioMLBase` class:
 
+- `self.label_config` - Returns the [Label Studio labeling config](setup) as XML string.
```
Comment (niklub) [inline docs/source/guide/ml_create.md]:
```suggestion
- `self.label_interface` - Returns the Label Studio Label Interface object that contains all information about the labeling task
```

File: docs/source/guide/ml_create.md [hunk 27]
```diff
@@ -177,30 +114,72 @@ def fit(self, event, data, **kwargs):
...
+- `self.parsed_label_config` - Returns the [Label Studio labeling config](setup) as JSON.
```
Comment (niklub) [inline docs/source/guide/ml_create.md]: no longer needed

File: docs/source/guide/ml_create.md [hunk 28]
```diff
@@ -177,30 +114,72 @@ def fit(self, event, data, **kwargs):
...
+label-studio-ml start my_ml_backend
+```
+
+### Modify the port
+
+To modify the port, use the `-p` parameter:
+
+```bash
+label-studio-ml start my_ml_backend -p 9091
+```
```
Comment (niklub) [inline docs/source/guide/ml_create.md]:
```suggestion
### Modify the host and port

To modify the host and port, use the following command line parameters:

```bash
label-studio-ml start my_ml_backend -p 9091 --host 0.0.0.0
```
```

File: docs/source/guide/project_settings.md [hunk 29]
```diff
@@ -32,41 +32,58 @@ The labeling interface is the central configuration point for projects. This det
...
-<dd>
+Use the drop-down menu to select the predictions source. For example, you can select a [connected model](#Model) or a set of [predictions](#Predictions). 
+
+</dd>
+
+</dl>
+
+## Model
+
+Click **Add Model** to connect an machine learning (ML) backend to your project. For more information on connecting a model, see [Machine learning integration](ml).
```
Comment (niklub) [inline docs/source/guide/project_settings.md]:
```suggestion
Click **Connect Model** to connect a machine learning (ML) backend to your project. For more information on connecting a model, see [Machine learning integration](ml).
```

File: docs/source/guide/project_settings.md [hunk 30]
```diff
@@ -32,41 +32,58 @@ The labeling interface is the central configuration point for projects. This det
...
+| **Start model training on annotation submission**         | Triggers the connected ML backend to start the training process each time an annotation is created or updated. <br /><br />This is part of an [active learning loop](active_learning) where the model can be continuously improved as new annotations are added to the dataset. When this setting is enabled, the ML backend's `fit()` method is called, allowing the model to learn from the most recent annotations and potentially improve its predictions for subsequent tasks.   |
+| **Interactive preannotations**         | (Available when creating or editing a model connection)<br /><br />Enable this option to allow the model to assist with the labeling process by providing real-time predictions or suggestions as annotators work on tasks.  <br /><br />In other words, as you interact with data (for example, by drawing a region on an image, highlighting text, or asking an LLM a question), the ML backend receives this input and returns predictions based on it.   |
```
Comment (niklub) [inline docs/source/guide/project_settings.md]: please add link: [more about interactive preannotations...]()

File: docs/source/guide/project_settings_lse.md [hunk 31]
```diff
@@ -376,33 +387,30 @@ To assign a project-level role, first add the person to your project. Once added
...
 
-## Machine learning
-
-Click **Add Model** to connect an machine learning (ML) backend to your project. For more information about using ML backends, see [Machine learning integration](ml).
+## Model
 
-<dl>
-
-<dt>ML-Assisted Labeling</dt>
+Click **Add Model** to connect an machine learning (ML) backend to your project. For more information on connecting a model, see [Machine learning integration](ml).
```
Comment (niklub) [inline docs/source/guide/project_settings_lse.md]:
```suggestion
Click **Connect Model** to connect a machine learning (ML) backend to your project. For more information on connecting a model, see [Machine learning integration](ml).
```

File: docs/source/guide/project_settings_lse.md [hunk 32]
```diff
@@ -376,33 +387,30 @@ To assign a project-level role, first add the person to your project. Once added
...
+| **Start model training on annotation submission**         | Triggers the connected ML backend to start the training process each time an annotation is created or updated. <br /><br />This is part of an [active learning loop](active_learning) where the model can be continuously improved as new annotations are added to the dataset. When this setting is enabled, the ML backend's `fit()` method is called, allowing the model to learn from the most recent annotations and potentially improve its predictions for subsequent tasks.   |
+| **Interactive preannotations**         | (Available when creating or editing a model connection)<br /><br />Enable this option to allow the model to assist with the labeling process by providing real-time predictions or suggestions as annotators work on tasks.  <br /><br />In other words, as you interact with data (for example, by drawing a region on an image, highlighting text, or asking an LLM a question), the ML backend receives this input and returns predictions based on it.   |
```
Comment (niklub) [inline docs/source/guide/project_settings_lse.md]: same note about the link to the interactive labeling guide

File: docs/source/guide/project_settings_lse.md [hunk 33]
```diff
@@ -376,33 +387,30 @@ To assign a project-level role, first add the person to your project. Once added
...
+| **Start Training**         | Manually initiate training. Use this action if you want to control when the model training occurs, such as after a specific number of annotations have been collected or at certain intervals.  |
+| **Send Test Request**         | (Available from the overflow menu next to the connected model)<br /><br />Use this for troubleshooting and sending a test resquest to the connected model.   |
+| **Edit**         | Edit the model name, URL, and parameters. For more information, see [Connect a model to Label Studio](ml#Connect-a-model-to-Label-Studio). |
+| **Delete**         | Remove the connection to the model. |
 
-</dd>
+## Predictions
 
-</dl>
+From here you can view predictions that have been imported or generated through training. For more information, see [Import pre-annotated data into Label Studio](predictions).
```
Comment (niklub) [inline docs/source/guide/project_settings_lse.md]: `...generated through training...` - that's not correct. I assume the idea is about "generated in batch via 'Retrieve Predictions' or 'Batch Predictions'" action called when selecting the task in Data Manager. We need to fix this elsewhere, and ensure we have the workflow described, and the link provided

**Review by niklub — CHANGES_REQUESTED** [review]

File: docs/source/guide/ml.md (hunk 6 above)
Comment (caitlinwheeless) [reply docs/source/guide/ml.md]: I had that table here, but then you kept updating it! I can add it back once it's more finalized.

File: docs/source/guide/ml.md (hunk 5 above)
Comment (caitlinwheeless) [reply docs/source/guide/ml.md:28]: See line 21 - added list

File: docs/source/guide/project_settings_lse.md (hunk 33 above)
Comment (caitlinwheeless) [reply docs/source/guide/project_settings_lse.md]: See changes - not sure if I captured this correctly

File: docs/source/guide/project_settings_lse.md [hunk 34]
```diff
@@ -376,33 +387,30 @@ To assign a project-level role, first add the person to your project. Once added
...
+| **Start Training**         | Manually initiate training. Use this action if you want to control when the model training occurs, such as after a specific number of annotations have been collected or at certain intervals.  |
+| **Send Test Request**         | (Available from the overflow menu next to the connected model)<br /><br />Use this for troubleshooting and sending a test resquest to the connected model.   |
+| **Edit**         | Edit the model name, URL, and parameters. For more information, see [Connect a model to Label Studio](ml#Connect-a-model-to-Label-Studio). |
+| **Delete**         | Remove the connection to the model. |
 
-</dd>
+## Predictions
 
-</dl>
+From here you can view predictions that have been imported or generated when executing the **Batch Predictions** action from the Data Manager. For more information on using predictions, see [Import pre-annotated data into Label Studio](predictions).
```
Comment (caitlinwheeless) [inline docs/source/guide/project_settings_lse.md:413]: Updated - double check if correct

File: docs/source/guide/ml.md [hunk 35]
```diff
@@ -114,39 +175,35 @@ If you want to retrieve predictions manually for a list of tasks **using only an
...
+Smart tools duplicate the tools that you have set up to use through your labeling configuration (e.g. rectangles) to allow interaction with the ML backend. Smart tools are currently for Rectangle, Ellipse, Polygon, Keypoint, and Brush tags. 
 
-This option is supported for Rectangle, Ellipse, Polygon, Keypoint, and Brush tags. See the [tag documentation](/tags). 
-   
-1. For your project, open **Settings > Labeling Interface**.
-2. Click **Code** to view the XML labeling configuration.
-3. For the relevant tag type that you want to use to generate predictions with your ML backend, add the `smart="true"` parameter. For example: 
+Smart tools are dynamic and can change behavior or appearance based on the context, leveraging machine learning predictions. For example, smart tools can automatically detect and suggest annotations for objects in images, which annotators can then review and refine. This can significantly speed up the labeling process, especially when dealing with large datasets.
```
Comment (caitlinwheeless) [inline docs/source/guide/ml.md:211]: Definitely read through this new section

File: docs/source/guide/ml.md (hunk 6 above)
Comment (niklub) [reply docs/source/guide/ml.md]: it is finalized now

File: docs/source/guide/ml.md [hunk 36]
```diff
@@ -114,39 +175,35 @@ If you want to retrieve predictions manually for a list of tasks **using only an
...
+
+Smart tools use `context` to adapt their behavior based on the current state of the labeling environment and the specific needs of the task at hand. For instance, they might alter their functionality depending on the type of data being labeled or the particular region of interest within a task. W
+
+hen smart tools are enabled for a labeling task, they can interact with the ML backend by sending data to the `/predict` endpoint to receive predictions. These predictions are then used to provide interactive pre-annotations within the labeling interface. For more information, see [Support interactive pre-annotations in your ML backend](ml_create#Support-interactive-pre-annotations-in-your-ML-backend).
```
Comment (niklub) [inline docs/source/guide/ml.md]:
```suggestion
Smart tools use `context` to adapt their behavior based on the current state of the labeling environment and the specific needs of the task at hand. For instance, they might alter their functionality depending on the type of data being labeled or the particular region of interest within a task. When smart tools are enabled for a labeling task, they can interact with the ML backend by sending data to the `/predict` endpoint to receive predictions. These predictions are then used to provide interactive pre-annotations within the labeling interface. For more information, see [Support interactive pre-annotations in your ML backend](ml_create#Support-interactive-pre-annotations-in-your-ML-backend). 
```

File: docs/source/guide/ml.md [hunk 37]
```diff
@@ -114,39 +175,35 @@ If you want to retrieve predictions manually for a list of tasks **using only an
...
 
-1. In the Label Studio UI, open the project that you want to use with your ML backend.
-2. Select **Settings > Machine Learning**.
-3. For the connected model you wish to enable, click **Edit** in the overflow menu.
-4. Enable **Use for interactive preannotation**.
-5. Click **Validate and Save**.
+#### Smart tools
 
-For image labeling, you can update your labeling configuration to include `smart="true"` option for the type of labeling you're performing. Smart tools appear by default if auto-annotation is enabled. If you only want the smart option to appear and don't want to perform manual labeling at all, use `smartOnly="true"`. 
+Smart tools duplicate the tools that you have set up to use through your labeling configuration (e.g. rectangles) to allow interaction with the ML backend. Smart tools are currently for Rectangle, Ellipse, Polygon, Keypoint, and Brush tags.
```
Comment (niklub) [inline docs/source/guide/ml.md]: shall we provide exact labeling tags instead?

File: docs/source/guide/project_settings_lse.md (hunk 34 above)
Comment (niklub) [reply docs/source/guide/project_settings_lse.md:413]: it is called `Batch Predictions` in LSE and `Retrieve Predictions` in LS

File: docs/source/guide/ml.md (hunk 6 above)
Comment (caitlinwheeless) [reply docs/source/guide/ml.md]: Added table

**Review by niklub — CHANGES_REQUESTED** [review]

File: docs/source/guide/project_settings_lse.md (hunk 34 above)
Comment (caitlinwheeless) [reply docs/source/guide/project_settings_lse.md:413]: This is the LSE version of the Project Settings page, so I think it's correct. I don't mention the action in the LS version - I will update there.

File: docs/source/guide/ml.md (hunk 37 above)
Comment (caitlinwheeless) [reply docs/source/guide/ml.md]: Added tag formatting and links to the appropriate tag pages

**Review by niklub — APPROVED** [review]

**Pull request merged**